common/djangoapps/student/views/management.py

```diff
@@ -778,7 +789,10 @@ def confirm_email_change(request, key):
         u_prof.save()
         # Send it to the old email...
         try:
-            ace.send(msg)
+            if is_testing_environment():
+                ace.send(msg)
+            else:
+                compose_and_send_adg_update_email_confirmation(user, message_context)
```

Comment:
These lines are repeated from line 808-811. Can we somehow remove this duplication, by introducing a new function?

Reply:
These are written in edX core code and the only reason I write those is that edX tests should avoid this code, if we introduce a function here than that function will be called in edX unit tests and thus may produce anomalies. If it's still a concern I would do that.

Reply:
can't we mock that function easily? If it requires much extra effort, we can keep things as it is.